assert/assertions.go

```diff
@@ -632,34 +632,20 @@ func NotNil(t TestingT, object interface{}, msgAndArgs ...interface{}) bool {
 	return Fail(t, "Expected value not to be nil.", msgAndArgs...)
 }
 
-// containsKind checks if a specified kind in the slice of kinds.
-func containsKind(kinds []reflect.Kind, kind reflect.Kind) bool {
-	for i := 0; i < len(kinds); i++ {
-		if kind == kinds[i] {
-			return true
-		}
-	}
-
-	return false
-}
-
 // isNil checks if a specified object is nil or not, without Failing.
 func isNil(object interface{}) bool {
 	if object == nil {
 		return true
 	}
 
 	value := reflect.ValueOf(object)
-	kind := value.Kind()
-	isNilableKind := containsKind(
-		[]reflect.Kind{
-			reflect.Chan, reflect.Func,
-			reflect.Interface, reflect.Map,
-			reflect.Ptr, reflect.Slice, reflect.UnsafePointer},
-		kind)
-
-	if isNilableKind && value.IsNil() {
-		return true
+	switch value.Kind() {
+	case
+		reflect.Chan, reflect.Func,
```

Comment:
Also, more dirty option, but protect us from supporting of new possible nillable types:

```go
// isNil checks if a specified object is nil or not, without Failing.
func isNil(object interface{}) (result bool) {
	if object == nil {
		return true
	}
	
	defer func() {
		if err := recover(); err != nil {
			result = false
		}
	}()
	return reflect.ValueOf(object).IsNil()
}
```

Reply:
Interesting. However I have concerns about performance because of panic/defer. That would require a benchmark.

But if the goal is to be future-proof, at least that alternative implementation would be interesting for validation of `assert.Nil`.

Reply:
As I don't think that that extra safety net is necessary I'm keeping my original version.